src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -93,27 +93,39 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 	}
 	
 	@Override
-	public Collection<AddOnInfoSummary> search(AddOnType type, String query) throws IOException {
+	public Collection<AddOnInfoSummary> search(AddOnType type, String query, String tag) throws IOException {
 		
 		BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
 		if (type != null) {
-			//Gets all modules of the type specified by variable type
+			//Exact match on type
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
+		if (tag != null) {
+			
+		SearchResult result = client.execute(new Search.Builder(new SearchSourceBuilder()
+				.size(SEARCH_SIZE)
+				.query(QueryBuilders.matchQuery("tags", tag)).toString())
+				.addIndex(AddOnInfoAndVersions.ES_INDEX)
+				.build());
+		return result.getHits(AddOnInfoAndVersions.class).stream()
+				.map(sr -> new AddOnInfoSummary(sr.source))
+				.collect(Collectors.toList());
+
```

Comment:
Also, please try to keep the formatting clean. E.g. there should be no extraneous blank lines. (It's fine to have an intentional blank line to make the code more readable, but that should almost never be the first or last line of a block.